hibernate-core/src/main/java/org/hibernate/engine/jdbc/internal/BasicFormatterImpl.java

```diff
@@ -110,127 +110,43 @@ public String perform() {
 
```

Comment:
It might be an idea to put `Enum` to further use to improve the various `Set` or `Map` looking up. Take `BEGIN_CLAUSES` as an example. We might first define its own enum class as:
```
private enum BeginClause {
    ... ...
}
```
Then replace `Set` or `Map` with `EnumMap` which is compact and cannot be beat in terms of performance for basically it is array based. The idea might apply for we have many such small-sized enum candidates for the static constants.
Not 100% sure of my comment's validity though.